Python/Product/Analysis/Analyzer/InterpreterScope.cs

```diff
@@ -290,8 +290,11 @@ public InterpreterScope(AnalysisValue av, InterpreterScope outerScope)
             _nodeScopes.Clear();
         }
 
-        public virtual IAnalysisSet AddNodeValue(Node node, IAnalysisSet variable) {
-            return _nodeValues[node] = variable;
+        public virtual IAnalysisSet AddNodeValue(Node node, NodeValueKind kind, IAnalysisSet variable) {
+            NodeValue next;
+            _nodeValues.TryGetValue(node, out next);
+            _nodeValues[node] = new NodeValue(kind, variable, next);
```

Comment:
That's actually the pattern here...  There's a GetOrMakeNodeValue which everyone uses.  AddNodeValue exists solely for consumption by GetOrMakeNodeValue.  Unfortunately we can't quite get the visibility right on this because derived scopes (comprehension, and is instance) override this and forward it to the real scope.  But if we declare it as protected we can't dispatch to the outer scope one on the base type.